src/mon/Monitor.cc

```diff
@@ -5334,6 +5334,16 @@ void Monitor::count_metadata(const string& field, Formatter *f)
   f->close_section();
 }
 
+void Monitor::check_daemon_versions(map<string,map<int, std::pair<int,string>>, std::greater<std::string>> &versions){
+  int i = 0;
+  for (auto& p : mon_metadata) {
+    auto q = p.second.find("ceph_version");
+    auto l = p.second.find("hostname");
+    versions[q->second][i] = std::make_pair(p.first, l->second);
```

Comment:
```suggestion
  for (auto& [rank, metadata] : mon_metadata) {
    auto& pv = metadata.at("ceph_version");
    auto& ph = metadata.at("hostname");
    versions[pv.second][i] = std::make_pair(rank, ph.second);
```

would be easier to read.

Reply:
@bpatrick If you like to look at the latest code you can see if your suggestion may apply partially.

Reply:
I made the change in the new location where similar code exists.